src/coldfront_plugin_api/serializers.py

```diff
@@ -23,28 +28,72 @@ def get_status(self, obj: Project) -> str:
         return obj.status.name
 
 
+class AllocationAttributeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = AllocationAttribute
+        fields = ["allocation_attribute_type", "value"]
+
+    allocation_attribute_type = serializers.SlugRelatedField(
+        read_only=False,
+        slug_field="name",
+        queryset=AllocationAttributeType.objects.all(),
+    )
+    value = serializers.CharField(read_only=False)
+
+
 class AllocationSerializer(serializers.ModelSerializer):
     class Meta:
         model = Allocation
-        fields = ["id", "project", "description", "resource", "status", "attributes"]
+        fields = [
+            "id",
+            "project",
+            "project_id",
+            "description",
+            "resource",
+            "resources_id",
+            "status",
+            "allocationattribute_set",
+        ]
+        read_only_fields = ["status"]
 
+    # Seperate serializer fields for reading and writing projects and resources
     resource = serializers.SerializerMethodField()
-    project = ProjectSerializer()
-    attributes = serializers.SerializerMethodField()
-    status = serializers.SerializerMethodField()
+    resources_id = serializers.PrimaryKeyRelatedField(
+        write_only=True, queryset=Resource.objects.all(), source="resources", many=True
+    )
+
+    project = ProjectSerializer(read_only=True)
+    project_id = serializers.PrimaryKeyRelatedField(
+        write_only=True,
+        queryset=Project.objects.all(),
+        source="project",
+    )
+
+    allocationattribute_set = AllocationAttributeSerializer(many=True, required=False)
 
     def get_resource(self, obj: Allocation) -> dict:
         resource = obj.resources.first()
         return {"name": resource.name, "resource_type": resource.resource_type.name}
 
-    def get_attributes(self, obj: Allocation):
-        attrs = AllocationAttribute.objects.filter(allocation=obj)
-        return {
-            a.allocation_attribute_type.name: obj.get_attribute(
-                a.allocation_attribute_type.name
-            )
-            for a in attrs
-        }
-
     def get_status(self, obj: Allocation) -> str:
         return obj.status.name
+
+    def create(self, validated_data):
+        allocation = Allocation.objects.create(
+            project=validated_data["project"],
+            status=AllocationStatusChoice.objects.get(
+                name="Active"
```

Comment:
Good question.

`New` would be the safe choice.

Otherwise allowing it to be provided as part of the API query during creation and default to `New` if missing.

Reply:
Do not send a signal when an allocation is created in any state. If someone modifies an allocation to change status, then do send a signal. So if `New -> Active = activate` or `Active -> Denied|Revoked = deactivate`